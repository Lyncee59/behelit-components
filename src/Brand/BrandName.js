/* eslint:disable */
import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Text } from '../Typography'
import { theme } from '../Tools/interpolation'

const Wrapper = styled(Text).attrs({ weight: 700 })`
  color: ${theme('brandNameColor')};
`

const BrandName = (props) => (
  <Wrapper {...props}>
    Behelit
  </Wrapper>
)

BrandName.propTypes = {
  size: PropTypes.string
}

BrandName.defaultProps = {
  size: '30px'
}

export default BrandName
