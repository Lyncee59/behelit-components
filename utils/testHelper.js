import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)