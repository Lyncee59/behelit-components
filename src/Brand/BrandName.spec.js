import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import BrandName from './BrandName'

describe('BrandName', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<BrandName />)

    expect(tree.find(BrandName)).toMatchSnapshot()
  })

  it('should render correctly with size', () => {
    const tree = mountWithTheme(<BrandName size="100px" />)

    expect(tree.find(BrandName)).toMatchSnapshot()
  })
})
