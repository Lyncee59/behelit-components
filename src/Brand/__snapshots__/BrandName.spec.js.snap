// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandName should render correctly 1`] = `
.c1 {
  font-family: 'Roboto',sans-serif;
  font-weight: 700;
  font-size: 30px;
  text-transform: none;
  font-style: normal;
  color: rgba(0,0,0,0.80);
  opacity: 1;
}

.c0 {
  color: #990000;
}

<BrandName
  size="30px"
>
  <Styled(Text)
    size="30px"
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "weight": 700,
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "Styled(Text)",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      forwardedRef={null}
      size="30px"
    >
      <Text
        capitalize={false}
        className="c0"
        italic={false}
        opacity={1}
        size="30px"
        uppercase={false}
        weight={700}
      >
        <styled.span
          capitalize={false}
          className="c0"
          italic={false}
          opacity={1}
          size="30px"
          uppercase={false}
          weight={700}
        >
          <StyledComponent
            capitalize={false}
            className="c0"
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  font-family: ",
                    [Function],
                    ";
  font-weight: ",
                    [Function],
                    ";
  font-size: ",
                    [Function],
                    ";
  text-transform: ",
                    [Function],
                    ";
  font-style: ",
                    [Function],
                    ";
  color: ",
                    [Function],
                    ";
  opacity: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.span",
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            italic={false}
            opacity={1}
            size="30px"
            uppercase={false}
            weight={700}
          >
            <span
              className="c0 c1"
              opacity={1}
              size="30px"
            >
              Behelit
            </span>
          </StyledComponent>
        </styled.span>
      </Text>
    </StyledComponent>
  </Styled(Text)>
</BrandName>
`;

exports[`BrandName should render correctly with size 1`] = `
.c1 {
  font-family: 'Roboto',sans-serif;
  font-weight: 700;
  font-size: 100px;
  text-transform: none;
  font-style: normal;
  color: rgba(0,0,0,0.80);
  opacity: 1;
}

.c0 {
  color: #990000;
}

<BrandName
  size="100px"
>
  <Styled(Text)
    size="100px"
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "weight": 700,
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "Styled(Text)",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      forwardedRef={null}
      size="100px"
    >
      <Text
        capitalize={false}
        className="c0"
        italic={false}
        opacity={1}
        size="100px"
        uppercase={false}
        weight={700}
      >
        <styled.span
          capitalize={false}
          className="c0"
          italic={false}
          opacity={1}
          size="100px"
          uppercase={false}
          weight={700}
        >
          <StyledComponent
            capitalize={false}
            className="c0"
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  font-family: ",
                    [Function],
                    ";
  font-weight: ",
                    [Function],
                    ";
  font-size: ",
                    [Function],
                    ";
  text-transform: ",
                    [Function],
                    ";
  font-style: ",
                    [Function],
                    ";
  color: ",
                    [Function],
                    ";
  opacity: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.span",
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            italic={false}
            opacity={1}
            size="100px"
            uppercase={false}
            weight={700}
          >
            <span
              className="c0 c1"
              opacity={1}
              size="100px"
            >
              Behelit
            </span>
          </StyledComponent>
        </styled.span>
      </Text>
    </StyledComponent>
  </Styled(Text)>
</BrandName>
`;
